src/GraphQL/Types/Scalars/ULongGraphType.cs

```diff
@@ -21,13 +21,20 @@ public class ULongGraphType : ScalarGraphType
         /// <inheritdoc/>
         public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(ulong));
 
+        /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value switch
         {
-            ULongValue _ => true,
             IntValue intValue => intValue.Value >= 0,
             LongValue longValue => longValue.Value >= 0,
             BigIntValue bigIntValue => bigIntValue.Value >= 0 && bigIntValue.Value <= ulong.MaxValue,
             _ => false
         };
+
+        /// <inheritdoc/>
+        public override IValue ToAST(object value) => value switch
```

Comment:
The same comment as https://github.com/graphql-dotnet/graphql-dotnet/pull/2321/files#r584122575

Maybe try to use "the lowest possible" value nodes ?  I'm not sure though. Consider someone called `new DecimalGraphType().ToAst(1)`. What is better to return ? `DecimalValue`? `IntValue`? `LongValue`? I'm leaning towards `DecimalValue`.

But in the case of unsigned types, I would still not "promote" value node to the "next level", i.e. I think that `new ULongGraphType().ToAst(1UL)` should return `new IntValue((int)1UL)`.

Reply:
But why? Really we "should" remove `IntValue` and `LongValue`, and only retain `BigIntValue`, which will most closely represent the original AST.  However, we have `IntValue` and `LongValue` as speed optimizations.

Reply:
The integer scalars now all interpret all versions of numeric integer AST representations.

Reply:
Good.

Reply:
> But in the case of unsigned types, I would still not "promote" value node to the "next level", i.e. I think that new ULongGraphType().ToAst(1UL) should return new IntValue((int)1UL).

I'm not sure. Let's keep it as is.